video2dataset/worker.py

```diff
@@ -125,9 +125,16 @@ def data_generator():
         )
         oom_sample_per_shard = math.ceil(math.log10(self.number_sample_per_shard))
 
+        # for exception handling
```

Comment:
What about doing this inside the data reader instead

Reply:
There can be many potential implementations of the data reader but they would all report to the worker